internal/httpclient/client_socks5_test.go

```
@@ -0,0 +1,217 @@
+// whose HTTPS_PROXY is a plain http(s) proxy, while HTTP_PROXY happens to be
+// a SOCKS5 proxy. net/http.Transport dials the CONNECT tunnel's proxy
+// address (not the request's target) through DialContext for this request,
+// so the dial decision must come from the request's own scheme (https,
+// which resolves to a plain proxy) -- never re-derived from that proxy
+// address as if it were something to look up HTTP_PROXY for.
+func TestDialDecisionIsNotReresolvedFromProxyHopAddress(t *testing.T) {
+	clearProxyEnv(t)
+	t.Setenv("HTTPS_PROXY", "http://real-proxy.example:8080")
+	t.Setenv("HTTP_PROXY", "socks5h://bogus-proxy.invalid:1080")
+
+	req, err := http.NewRequest(http.MethodGet, "https://example.com/stream", nil)
```

golangci-lint (2.13.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)